OpenKeychain/src/main/java/org/sufficientlysecure/keychain/remote/SshAuthenticationService.java

```diff
@@ -141,12 +147,22 @@ private Intent authenticate(Intent data) {
         CachedPublicKeyRing cachedPublicKeyRing = mKeyRepository.getCachedPublicKeyRing(masterKeyId);
 
         long authSubKeyId;
+        int authSubKeyAlgorithm;
+        CanonicalizedPublicKey authSubPublicKey = null;
         try {
             // get first usable subkey capable of authentication
             authSubKeyId = cachedPublicKeyRing.getSecretAuthenticationId();
+            // needed for encoding the resulting signature
+            authSubKeyAlgorithm = getPublicKey(masterKeyId).getAlgorithm();
+            if (authSubKeyAlgorithm == PublicKeyAlgorithmTags.ECDSA) {
+                authSubPublicKey = getPublicKey(masterKeyId);
```

Comment:
using implicit knowledge here that `SshSignature` only requires the key object for ECDSA couples these classes tighter than necessary. this should go away if you juts pass algorithm + curveOid

Reply:
Yes, thanks. SshSignature was rather badly designed. It's now SshSignatureConverter with a static methods for converting ECDSA signatures and a static method for the rest.